bin/impl/fluo_deploy/util.py

```diff
@@ -108,27 +109,49 @@ def setup_boto(lib_dir):
     print "Finished downloading Boto"
   sys.path.insert(0, boto_lib_dir)
 
-def parse_args(input=None):
+def parse_args(hosts_dir, input=None):
   parser = OptionParser(
-            usage="fluo-deploy [options] <action> <cluster_name>"
+            usage="fluo-deploy [options] <action>"
             + "\n\nwhere <action> can be:\n  launch - Launch cluster in EC2\n  status - Check status of EC2 cluster\n"
-            + "  setup - Setup Fluo and its dependencies on cluster\n  ssh - SSH to cluster leader node\n"
-            + "  kill - Kill Fluo and its dependencies on cluster\n  terminate - Terminate EC2 cluster",
+            + "  setup - Setup Fluo and its dependencies on cluster\n  config - Print configuration for that cluster\n"
+            + "  ssh - SSH to cluster leader node\n  kill - Kill Fluo and its dependencies on cluster\n"
+            + "  terminate - Terminate EC2 cluster",
             add_help_option=False)
-  parser.add_option(
-    "-h", "--help", action="help",
-    help="Show this help message and exit")
+  parser.add_option("-c", "--cluster", dest="cluster", help="Specifies cluster")
+  parser.add_option("-p", "--property", dest="property", help="Specifies property to print (if using 'config' action).  Set to 'all' to print every property")
+  parser.add_option("-h", "--help", action="help", help="Show this help message and exit")
 
   if input:
     (opts, args) = parser.parse_args(input)
   else:
     (opts, args) = parser.parse_args()
 
-  if len(args) != 2:
-    parser.print_help()
-    sys.exit(1)
+  if len(args) == 0:
+    print "ERROR - You must specify on action"
+    return
+  elif len(args) > 1:
+    print "ERROR - Too many arguments given"
+    return
+  action = args[0]
 
-  (action, cluster_name) = args
+  if action == 'launch' and not opts.cluster:
```

Comment:
it should come before.  the launch command shouldn't read cluster name from conf/hosts as it is launching a new cluster and not working with an existing cluster.